storage/diff_scheme/diff_manager.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "storage/diff_scheme/diff_types.hpp"
+#include "storage/index.hpp"
+
+#include "base/observer_list.hpp"
+#include "base/thread_checker.hpp"
+#include "base/worker_thread.hpp"
+
+#include <functional>
+#include <mutex>
+#include <utility>
+
+namespace storage
+{
+namespace diffs
+{
+struct ApplyDiffParams
+{
+  string m_diffReadyPath;
+  TLocalFilePtr m_diffFile;
+  TLocalFilePtr m_oldMwmFile;
+};
+
+class Observer
+{
+public:
+  virtual void OnDiffStatusReceived() {}
```

Comment:
virtual dtor too, please. Also, why not to make a pure OnDiffStatusReceived method?